#! /bin/sh /usr/share/dpatch/dpatch-run
## 05.configure.py_prog_prefer.dpatch.dpatch by Sven Hoexter <sven@timegate.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: configure.py should prefer some special programs from the packages 
## DP: debianutils and xdg-open

@DPATCH@

--- lyx-1.4.2svn/lib/configure.py-orig	2006-07-11 12:40:53.000000000 +0200
+++ lyx-1.4.2svn/lib/configure.py	2006-07-11 12:38:23.000000000 +0200
@@ -209,7 +209,7 @@
   checkViewer('a FEN viewer and editor', ['xboard -lpf $$i -mode EditPosition'],
     rc_entry = [r'\Format fen        fen     FEN                    "" "%%"	"%%"'])
   #
-  path, iv = checkViewer('a raster image viewer', ['xv', 'kview', 'gimp'])
+  path, iv = checkViewer('a raster image viewer', ['xdg-open','xv', 'kview', 'gimp'])
   path, ie = checkViewer('a raster image editor', ['gimp'])
   addToRC(r'''\Format bmp        bmp     BMP                    "" "%s"	"%s"
 \Format gif        gif     GIF                    "" "%s"	"%s"
@@ -223,7 +223,7 @@
 \Format xpm        xpm     XPM                    "" "%s"	"%s"''' % \
     (iv, ie, iv, ie, iv, ie, iv, ie, iv, ie, iv, ie, iv, ie, iv, ie, iv, ie, iv, ie) )
   #
-  checkViewer('a text editor', ['xemacs', 'gvim', 'kedit', 'kwrite', 'kate', \
+  checkViewer('a text editor', ['sensible-editor','xemacs', 'gvim', 'kedit', 'kwrite', 'kate', \
     'nedit', 'gedit', 'notepad'],
     rc_entry = [r'''\Format asciichess asc    "Plain text (chess output)"  "" ""	"%%"
 \Format asciiimage asc    "Plain text (image)"         "" ""	"%%"
@@ -240,17 +240,17 @@
   #
   #checkProg('a Postscript interpreter', ['gs'],
   #  rc_entry = [ r'\ps_command "%%"' ])
-  checkViewer('a Postscript previewer', ['gv', 'ghostview -swap', 'kghostview'],
+  checkViewer('a Postscript previewer', ['xdg-open','kghostview', 'gv','ghostview -swap', 'evince'],
     rc_entry = [r'''\Format eps        eps     EPS                    "" "%%"	""
 \Format ps         ps      Postscript             t  "%%"	""'''])
   #
-  checkViewer('a PDF previewer', ['acrobat', 'acroread', 'gv', 'ghostview', \
-              'xpdf', 'kpdf', 'kghostview'],
+  checkViewer('a PDF previewer', ['xdg-open','xpdf', 'acroread', 'acrobat', 'gv', 'ghostview', \
+              'kpdf', 'kghostview','evince'],
     rc_entry = [r'''\Format pdf        pdf    "PDF (ps2pdf)"          P  "%%"	""
 \Format pdf2       pdf    "PDF (pdflatex)"        F  "%%"	""
 \Format pdf3       pdf    "PDF (dvipdfm)"         m  "%%"	""'''])
   #
-  checkViewer('a DVI previewer', ['xdvi', 'kdvi'],
+  checkViewer('a DVI previewer', ['xdg-open','kdvi', 'xdvi','xgdvi','tkdvi','evince'],
     rc_entry = [r'\Format dvi        dvi     DVI                    D  "%%"	""'])
   if ((os.name == 'nt' or sys.platform == 'cygwin') and
       checkProg('DVI to DTL converter', ['dv2dt']) != ['', ''] and
@@ -258,7 +258,7 @@
     # Windows only: DraftDVI
     addToRC(r'\Format dvi2       dvi     DraftDVI               ""	""	""')
   #
-  checkViewer('a HTML previewer', ['mozilla file://$$p$$i', 'netscape'],
+  checkViewer('a HTML previewer', ['sensible-browser file://$$p$$i','mozilla file://$$p$$i', 'netscape'],
     rc_entry = [r'\Format html       html    HTML                   H  "%%"	""'])
   #
   # entried that do not need checkProg
